### PR TITLE
fix: Initialize logging for crashpad

### PR DESCRIPTION
**Pull request by deepak1556**
#### Description of Change

On windows the default logging mode is [LOG_TO_FILE](https://source.chromium.org/chromium/chromium/src/+/master:base/logging.h;l=200) and crashpad has been operating under this mode. The default log file will be created under the [executable directory](https://source.chromium.org/chromium/chromium/src/+/master:base/logging.cc;l=233-249) but when the executable is installed under restrictive directories like `C:\Program Files` the app won't have permission to write log files here and thereby creating `debug.log` file under the [working directory](https://source.chromium.org/chromium/chromium/src/+/master:base/logging.cc;l=283-286)

Crbug - https://bugs.chromium.org/p/chromium/issues/detail?id=711159
Refs https://github.com/microsoft/vscode/issues/105191

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: fix debug.log files being created under working directory on windows

### CODE REVIEW

**Comment by deepak1556** [comment]
osx-x64 has failing crash reporter tests due to crash in handler,

```
[1029/101831.854852:FATAL:command_line.cc(237)] Check failed: current_process_commandline_. 
0   chrome_crashpad_handler             0x00000001016783a9 base::debug::CollectStackTrace(void**, unsigned long) + 9
1   chrome_crashpad_handler             0x000000010159e2c3 base::debug::StackTrace::StackTrace() + 19
2   chrome_crashpad_handler             0x00000001015b7155 logging::LogMessage::~LogMessage() + 261
3   chrome_crashpad_handler             0x00000001015b7dee logging::LogMessage::~LogMessage() + 14
4   chrome_crashpad_handler             0x000000010158efd1 base::CommandLine::ForCurrentProcess() + 65
5   chrome_crashpad_handler             0x00000001015b60c9 logging::BaseInitLoggingImpl_built_with_NDEBUG(logging::LoggingSettings const&) + 25
6   chrome_crashpad_handler             0x000000010157a6ed crashpad::HandlerMain(int, char**, std::__1::vector<std::__1::unique_ptr<crashpad::UserStreamDataSource, std::__1::default_delete<crashpad::UserStreamDataSource> >, std::__1::allocator<std::__1::unique_ptr<crashpad::UserStreamDataSource, std::__1::default_delete<crashpad::UserStreamDataSource> > > > const*) + 93
7   chrome_crashpad_handler             0x000000010116da2a CrashpadHandlerMain + 90
8   libdyld.dylib                       0x00007fff68357cc9 start + 1
```

**Comment by nornagon** [comment]
Is this a straight backport or is the patch modified?

**Comment by deepak1556** [comment]
Crashpad patch from https://chromium-review.googlesource.com/c/crashpad/crashpad/+/2425108
Base logging patch from https://chromium-review.googlesource.com/c/chromium/src/+/2490880

File: patches/chromium/crashpad-initialize-logging.patch [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+diff --git a/third_party/crashpad/crashpad/DEPS b/third_party/crashpad/crashpad/DEPS
+index 83995e0cdea91522c272415330c57af764d23163..7e83327aac2582e81a38720086a52832de58a37a 100644
+--- a/third_party/crashpad/crashpad/DEPS
++++ b/third_party/crashpad/crashpad/DEPS
+@@ -42,7 +42,7 @@ deps = {
+       '7bde79cc274d06451bf65ae82c012a5d3e476b5a',
+   'crashpad/third_party/mini_chromium/mini_chromium':
+       Var('chromium_git') + '/chromium/mini_chromium@' +
+-      '76a9bb7475f6217eaf108789246379d3972b4e6a',
++      '5fc64bfbf1c000161445c586de45e40464ff2314',
```
Comment (nornagon) [inline patches/chromium/crashpad-initialize-logging.patch:82]: Patches to DEPS don't work as the dep is synced before our patches are applied.

**Review by nornagon — APPROVED** [review]

File: patches/chromium/crashpad-initialize-logging.patch (hunk 1 above)
Comment (deepak1556) [reply patches/chromium/crashpad-initialize-logging.patch:82]: This can be ignored, mini_chromium is not used when crashpad is built inside chromium

**Pull request merged**